bot/exts/info/pep.py

```diff
@@ -1,168 +1,90 @@
 from datetime import datetime, timedelta
-from email.parser import HeaderParser
-from io import StringIO
-from typing import Dict, Optional, Tuple
+from typing import Optional
 
 from discord import Colour, Embed
 from discord.ext.commands import Cog, Context, command
 
 from bot.bot import Bot
-from bot.constants import Keys
 from bot.log import get_logger
-from bot.utils.caching import AsyncCache
 
 log = get_logger(__name__)
 
 ICON_URL = "https://www.python.org/static/opengraph-icon-200x200.png"
-BASE_PEP_URL = "https://peps.python.org/pep-"
-PEPS_LISTING_API_URL = "https://api.github.com/repos/python/peps/contents?ref=main"
-
-pep_cache = AsyncCache()
-
-GITHUB_API_HEADERS = {}
-if Keys.github:
-    GITHUB_API_HEADERS["Authorization"] = f"token {Keys.github}"
+PEP_API_URL = "https://peps.python.org/api/peps.json"
 
 
 class PythonEnhancementProposals(Cog):
     """Cog for displaying information about PEPs."""
 
     def __init__(self, bot: Bot):
         self.bot = bot
-        self.peps: Dict[int, str] = {}
-        # To avoid situations where we don't have last datetime, set this to now.
-        self.last_refreshed_peps: datetime = datetime.now()
+        self.peps: dict[int, dict[str, Optional[str]]] = {}
+        self.last_refreshed_peps = datetime.now()
 
     async def cog_load(self) -> None:
         """Carry out cog asynchronous initialisation."""
-        await self.refresh_peps_urls()
+        await self.refresh_pep_data()
+
+    async def refresh_pep_data(self) -> None:
+        """Refresh PEP data."""
+        # Putting this first should prevent any race conditions
+        self.last_refreshed_peps = datetime.now()
 
-    async def refresh_peps_urls(self) -> None:
-        """Refresh PEP URLs listing in every 3 hours."""
         # Wait until HTTP client is available
         await self.bot.wait_until_ready()
-        log.trace("Started refreshing PEP URLs.")
-        self.last_refreshed_peps = datetime.now()
 
-        async with self.bot.http_session.get(
-            PEPS_LISTING_API_URL,
-            headers=GITHUB_API_HEADERS
-        ) as resp:
+        log.trace("Started refreshing PEPs data.")
+        async with self.bot.http_session.get(PEP_API_URL) as resp:
             if resp.status != 200:
-                log.warning(f"Fetching PEP URLs from GitHub API failed with code {resp.status}")
+                log.warning(
+                    f"Fetching PEP data from PEP API failed with code {resp.status}"
+                )
                 return
-
             listing = await resp.json()
 
-        log.trace("Got PEP URLs listing from GitHub API")
+        for pep_num, pep_info in listing.items():
+            self.peps[int(pep_num)] = pep_info
```

Comment:
Forgive me for asking but can't you just write these two lines as:
```suggestion
        self.peps = listing
```
The keys of the dictionary will be strings in this case though.

Reply:
I think i'll leave it, as being an int and having this code naming `pep_num` and `pep_info` makes it a little clearer what the structure of `peps` is imo.

Reply:
Alrighty.